src/services/credentialEvents/index.ts

```diff
@@ -131,4 +168,12 @@ export default class CredentialEvents {
   private isDone(credential: Credential): boolean {
     return credential.state === 'done'
   }
+  private problemReportStringParser(string: string) {
+    const schema = z.object({
+      message: z.string(),
+      pinTries: z.number(),
+    })
```

Comment:
not quite what I meant by pull this out. The actual `schema` should be shared at a class level